server/cmd/mmctl/commands/root.go

```diff
@@ -108,6 +108,12 @@ var RootCmd = &cobra.Command{
 		}
 		quiet := viper.GetBool("quiet")
 		printer.SetQuiet(quiet)
+
+		perPage, err := cmd.Flags().GetInt("per-page")
+		if err == nil && perPage > MaxPageSize {
+			cmd.Flags().Set("per-page", fmt.Sprintf("%d", MaxPageSize))
```

Comment:
I wonder if we should change the flag value at all. The server will lower the value anyway. If, in the future, we increase the value on the server, we might want for `mmctl` to still send the requests with a higher `per-page` value.

What do you think?

Reply:
Ideally client and server get updated at the same time, then this part doesn't really matter.

If they get out of sync, the options are:
1. We artificially limit the page size to a smaller number within the client (current implementation)
2. The updated size limit gets used, but users see an incorrect page-size-limit warning.

I'm not sure which situation is preferable, but it's an easy switch if you guys prefer option 2.

Reply:
I'm leaning towards option 2 and relaxing the language a bit to something like `Per page value is greater than the maximum allowed. Mattermost might only return %d items.`

@BenCookie95 what do you think>

Reply:
Pushed